modules/nvidia_plugin/src/transformer/cuda_graph_transformer.cpp

```diff
@@ -35,34 +37,46 @@
 #include "transformations/op_conversions/convert_subtract.hpp"
 #include "transformations/op_conversions/mvn6_decomposition.hpp"
 #include "transformations/common_optimizations/reshape_prelu.hpp"
-#include "ngraph/opsets/opset10.hpp"
 
 using namespace ov::nvidia_gpu;
 
-void GraphTransformer::common_transform(const CUDA::Device& device,
-                                        const std::shared_ptr<ov::Model>& model,
-                                        const InferenceEngine::InputsDataMap& inputInfoMap,
-                                        const InferenceEngine::OutputsDataMap& outputsInfoMap,
-                                        const Configuration& config) const {
+void GraphTransformer::transform(const CUDA::Device& device,
+                                 std::shared_ptr<ov::Model>& model,
+                                 const Configuration& config) const {
     auto inference_precision = config.get_inference_precision();
     if (inference_precision == ov::element::f16 && !isHalfSupported(device)) {
-        throwIEException("Inference precision f16 is not supported by device!");
+        throw_ov_exception("Inference precision f16 is not supported by device!");
     }
     auto upscale_precision = [&]() -> bool {
         return !isHalfSupported(device) || inference_precision == ov::element::f32;
     };
     auto downscale_precision = [&]() -> bool {
         return isHalfSupported(device) && inference_precision == ov::element::f16;
     };
-    auto passConfig = std::make_shared<ov::pass::PassConfig>();
-    ov::pass::Manager manager{passConfig};
 
-    passConfig->enable<ov::pass::ConvertInterpolate1ToInterpolate4>();
-    passConfig->disable<ov::pass::MVN6Decomposition>();
+    // WA for ConvertPrecision which doesn't keep original precisions (CVS-111453)
+    // Store original precisions for inputs and outputs
+    // And restore them after transformations if ConvertPrecision was called
+    std::map<size_t, ov::element::Type> input_preprocess;
+    std::map<size_t, ov::element::Type> output_postprocess;
+    for (size_t i = 0; i < model->inputs().size(); i++) {
+        auto input = model->input(i);
+        auto target_inputs = input.get_target_inputs();
+        input_preprocess[i] = input.get_element_type();
+    }
+    for (size_t i = 0; i < model->get_output_size(); i++) {
```

Comment:
in both cycles we use different approach :)

Reply:
> in both cycles we use different approach :)

Model doesn't have `get_input_size` function, only `get_output_size`, if you meant this)